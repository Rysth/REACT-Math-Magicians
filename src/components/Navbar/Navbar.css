.navbar {
  background-color: var(--CL-primary);
  padding: 1em 0;
}

.navbar-system {
  text-align: center;
}

.navbar-list {
  list-style-type: none;
  padding: 0;
}

.navbar-item.separator {
  display: none;
  color: var(--CL-neutral-white);
}

.navbar-link {
  color: var(--CL-neutral-white);
  text-decoration: none;
}

@media (min-width: 768px) {
  .navbar-system {
    display: flex;
    justify-content: space-between;
    align-items: center;
    text-align: initial;
  }

  .navbar-list {
    display: flex;
    gap: 0.5em;
  }

  .navbar-item.separator {
    display: block;
  }
}
