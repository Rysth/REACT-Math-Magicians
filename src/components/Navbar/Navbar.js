import React from 'react';
import { Link } from 'react-router-dom';
import './Navbar.css';

function Navbar() {
  return (
    <div className="navbar ">
      <div className="navbar-system container">
        <div className="navbar-left">
          <h1 className="title">Math Magicians</h1>
          <h2 className="subtitle">John Palacios</h2>
        </div>
        <nav className="navbar-right">
          <ul className="navbar-list">
            <li className="navbar-item">
              <Link to="/" className="navbar-link">
                Home
              </Link>
            </li>
            <li className="navbar-item separator">|</li>
            <li className="navbar-item">
              <Link to="/calculator" className="navbar-link">
                Calculator
              </Link>
            </li>
            <li className="navbar-item separator">|</li>
            <li className="navbar-item">
              <Link to="/quote" className="navbar-link">
                Quote
              </Link>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  );
}

export default Navbar;
