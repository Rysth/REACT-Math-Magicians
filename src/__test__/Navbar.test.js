import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter to be used to work similar as a BrowserRouter to make the Link component work properly.
import Navbar from '../components/Navbar/Navbar';

describe('Navbar Component', () => {
  test('Should simulate a click to the Home Page', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    const homeLink = getByText('Home');
    fireEvent.click(homeLink);
    expect(homeLink).toHaveTextContent('Home');
  });

  it('Should create a Navbar Component Snapshot', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <Navbar />
        </MemoryRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
