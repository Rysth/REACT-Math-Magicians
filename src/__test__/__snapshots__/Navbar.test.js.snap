// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component Should create a Navbar Component Snapshot 1`] = `
<div
  className="navbar "
>
  <div
    className="navbar-system container"
  >
    <div
      className="navbar-left"
    >
      <h1
        className="title"
      >
        Math Magicians
      </h1>
      <h2
        className="subtitle"
      >
        John Palacios
      </h2>
    </div>
    <nav
      className="navbar-right"
    >
      <ul
        className="navbar-list"
      >
        <li
          className="navbar-item"
        >
          <a
            className="navbar-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="navbar-item separator"
        >
          |
        </li>
        <li
          className="navbar-item"
        >
          <a
            className="navbar-link"
            href="/calculator"
            onClick={[Function]}
          >
            Calculator
          </a>
        </li>
        <li
          className="navbar-item separator"
        >
          |
        </li>
        <li
          className="navbar-item"
        >
          <a
            className="navbar-link"
            href="/quote"
            onClick={[Function]}
          >
            Quote
          </a>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
